src/engine/MaxConstraint.cpp

```diff
@@ -408,28 +452,94 @@ List<PiecewiseLinearCaseSplit> MaxConstraint::getCaseSplits() const
         splits.append( getSplit( _f ) );
     }
 
+    if ( _eliminatedVariables ) { // todo check with GuyK (I didn't add tightenings)
+        PiecewiseLinearCaseSplit phaseOfEliminatedIsMax;
+        // TODO - GuyA - move up to the case when _f exists only?
```

Comment:
in the case f is an element (above), add also that f >= maxEliminatedValue if it exists. Then, your new "if" should only be triggered if f is not an element.